### ISSUE TITLE
class-signature rule joins EOL comment with super type when max_line_length = off

### ISSUE THREAD

**Issue by hugoncosta** [pinterest/ktlint#3293]
## Expected Behavior

When a class declaration has a comment between the colon and the super type, `ktlintFormat` should preserve the comment on its own line. The following code should remain unchanged (or at minimum, not produce invalid syntax):

```kotlin
class PrefixConverter(val prefix: String, val separator: String = "#") :
    // Generic types: F = without prefix, T = with prefix
    Converter<String, String> {
    override fun convertFrom(to: String): String = to.substring(prefix.length + separator.length)
    override fun convertTo(from: String): String = "$prefix$separator$from"
}
```

## Observed Behavior

`ktlintFormat` joins the comment line with the super type declaration, producing invalid Kotlin syntax:

```kotlin
class PrefixConverter(val prefix: String, val separator: String = "#") :
    // Generic types: F = without prefix, T = with prefix Converter<String, String> {
    override fun convertFrom(to: String): String = to.substring(prefix.length + separator.length)
    override fun convertTo(from: String): String = "$prefix$separator$from"
}
```

This results in a compilation error:
```
e: file:///src/Dynamo.kt:50:1 Syntax error: Expecting a top level declaration.
```

The `standard:class-signature` rule emits:
```
BugRepro.kt:3:5: Expected single space before the super type (standard:class-signature)
```

It then "fixes" this by replacing the whitespace before the super type with a single space, but doesn't account for the EOL_COMMENT node between the colon and the super type — causing the comment text to be concatenated with the code.

## Steps to Reproduce

1. Create a file `BugRepro.kt`:
```kotlin
class PrefixConverter(val prefix: String, val separator: String = "#") :
    // Generic types: F = without prefix, T = with prefix
    Converter<String, String> {
    override fun convertFrom(to: String): String = to.substring(prefix.length + separator.length)
    override fun convertTo(from: String): String = "$prefix$separator$from"
}

interface Converter<F, T> {
    fun convertFrom(to: T): F
    fun convertTo(from: F): T
}
```

2. Create `.editorconfig`:
```ini
[*.kt]
ktlint_code_style = intellij_idea
max_line_length = off
```

3. Run:
```bash
ktlint --format BugRepro.kt
```

4. Observe that line 2-3 are joined into a single line containing both the comment and the super type.

**Potential Root cause:** In `ClassSignatureRule.kt`, `fixWhitespacesInSuperTypeList` — when there's a single super type and `wrappedPrimaryConstructor == false`, the code calls `upsertWhitespaceBeforeMe(" ")` without checking for intervening EOL_COMMENT nodes. The `wrappedPrimaryConstructor == true` branch has a `takeUnless` guard for comments, but the `else` branch does not.

## Your Environment

* Version of ktlint used: 1.8.0
* Relevant parts of the `.editorconfig` settings: `ktlint_code_style = intellij_idea`, `max_line_length = off`
* Name and version of integration used: ktlint CLI (also reproducible via ktlint-gradle plugin with `maxRuleVersion = "1.8.0"`)
* Version of Gradle used: 9.4.1
* Operating System and version: Amazon Linux 2 (x86_64), also reproducible on macOS

**Comment by paul-dingemans** [comment]
Problem can be reproduced with simplified example below:
```
class Foo : // comment
    Bar {
    override fun fooBar() = "foobar"
}

interface Bar {
    fun fooBar(): String
}
```

Also this reproduces the problem in the `ktlint_official` code style as well. Neither is disabling the max line length necessary to reproduce the problem with code sample above.

**Comment by hugoncosta** [comment]
Thank you Paul, much appreciated!
